src/measure_theory/measure/measure_space.lean

```diff
@@ -1824,6 +1824,81 @@ begin
     (lt_of_le_of_lt (le_iff'.1 h _) (C.finite i)), C.spanning⟩⟩,
 end
 
+section disjointed
+
+variable [measurable_space α]
+
+/-- Given `S : μ.finite_spanning_sets_in {s | measurable_set s}`,
+`disjointed_finite_spanning_sets_in` provides a `finite_spanning_sets_in {s | measurable_set s}`
+such that its underlying sets are pairwise disjoint. -/
+def disjointed_finite_spanning_sets_in (μ : measure α)
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+   μ.finite_spanning_sets_in {s | measurable_set s} :=
+⟨disjointed S.set, measurable_set.disjointed S.set_mem,
+  λ n, lt_of_le_of_lt (measure_mono (disjointed_subset S.set n)) (S.finite _),
+  S.spanning ▸ Union_disjointed⟩
+
+lemma disjoint_disjointed_finite_spanning_sets_in {μ : measure α}
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+  pairwise (disjoint on (disjointed_finite_spanning_sets_in μ S).set) :=
+disjoint_disjointed _
+
+/-- Given a pair of σ-finite measure `μ`, `ν`, `unpair_finite_spanning_sets_in_left μ ν` provides
+an element of type `μ.finite_spanning_sets_in {s | measurable_set s}` such that its underlying
+sets is pairwise disjoint, finite with respect to both `μ` and `ν` and equal to the underlying
+sets of `unpair_finite_spanning_sets_in_right μ ν`.
+
+Note: this definition is not commutative since `n.unpair.1` does not necessarily equal
+`n.unpair.2`. -/
+def unpair_finite_spanning_sets_in_left
+  (μ ν : measure α) [sigma_finite μ] [sigma_finite ν] :
+  μ.finite_spanning_sets_in {s | measurable_set s} :=
+μ.disjointed_finite_spanning_sets_in $
+let S := μ.to_finite_spanning_sets_in in
+let T := ν.to_finite_spanning_sets_in in
+{ set := λ n : ℕ, S.set n.unpair.1 ∩ T.set n.unpair.2,
+  set_mem := λ n, (S.set_mem n.unpair.1).inter (T.set_mem n.unpair.2),
+  finite := λ n, lt_of_le_of_lt (measure_mono (set.inter_subset_left _ _)) (S.finite n.unpair.1),
+  spanning := by simp_rw [set.Union_unpair (λ i j, S.set i ∩ T.set j), ← set.inter_Union,
+                          ← set.Union_inter, S.spanning, T.spanning, set.inter_univ] }
+
+/-- Given a pair of σ-finite measure `μ`, `ν`, `unpair_finite_spanning_sets_in_right μ ν` provides
+an element of type `ν.finite_spanning_sets_in {s | measurable_set s}` such that its underlying
+sets is pairwise disjoint, finite with respect to both `μ` and `ν`, and equal to the underlying
+sets of `unpair_finite_spanning_sets_in_left μ ν`.
+
+Note: this definition is not commutative since `n.unpair.1` does not necessarily equal
+`n.unpair.2`. -/
+def unpair_finite_spanning_sets_in_right
+  (μ ν : measure α) [sigma_finite μ] [sigma_finite ν] :
+  ν.finite_spanning_sets_in {s | measurable_set s} :=
+ν.disjointed_finite_spanning_sets_in $
```

Comment:
Is it possible to combine the two results as
```suggestion
(ν ⊔ μ).disjointed_finite_spanning_sets_in $
```
I can't work out what sup on measures is actually defined as, but if the coercion of the sups is the sup of the coercions then this would be true